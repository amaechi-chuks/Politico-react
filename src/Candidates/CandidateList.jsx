import React from 'react';
import PartyLogo from '../assets/img/PartyLogo.png';
import '../assets/style/howItWorks.css';
import PdpLogo from '../assets/img/PdpLogo.png';

const CandidateList = () => {
  return (
    <div>
      <main>
        <section className="container reports-container">
          <h2 className="section-title">All Candidates</h2>

          <div className="column featured-cards">
            <div className="card report-card">
              <div className="section1 clearfix">
                <div className="col2 last">
                  <div className="grid clearfix">
                    <div className="col3 first">
                      <h1>
                        <img
                          className="party-logo"
                          src={PartyLogo}
                          alt="APC logo"
                        />
                      </h1>
                      <span>Following</span>
                    </div>
                    <div className="col3">
                      <h1>452</h1>
                      <span>Members</span>
                    </div>
                  </div>
                </div>
              </div>
              <strong>Party Name: </strong>
              <div className="report-status resolved">
                All Progressives Congress (APC)
              </div>
              <div className="participant">
                <strong>Aspiring Candidate: </strong>
              </div>
              <p className="report-user">
                <strong>Muhammadu Buhari</strong>
              </p>
              <strong>Political office: </strong>
              <p className="report-time">President</p>
              <strong>Party Headquarter:</strong>
              <p className="report-location">4b Wuse Abuja, Nigeria</p>
              <strong>Registered:</strong>
              <p className="report-title">21st Nov. 2018 10:00 am</p>
              <p className="report-media">Political Interest</p>
            </div>
            <div className="card report-card">
              <div className="section1 clearfix">
                <div className="col2 last">
                  <div className="grid clearfix">
                    <div className="col3 first">
                      <h1>
                        <img
                          className="party-logo"
                          src={PdpLogo}
                          alt="APC logo"
                        />
                      </h1>
                      <span>Following</span>
                    </div>
                    <div className="col3">
                      <h1>452</h1>
                      <span>Members</span>
                    </div>
                  </div>
                </div>
              </div>
              <strong>Party Name: </strong>
              <div className="report-status resolved">
                People Democratic Party (PDP)
              </div>
              <div className="participant">
                <strong>Aspiring Candidate: </strong>
              </div>
              <p className="report-user">
                <strong>Atiku Abubakar</strong>
              </p>
              <strong>Political office: </strong>
              <p className="report-time">President</p>
              <strong>Party Headquarter:</strong>
              <p className="report-location">4b Wuse Abuja, Nigeria</p>
              <strong>Registered:</strong>
              <p className="report-title">21st Nov. 2018 10:00 am</p>
              <p className="report-media">Political Interest</p>
            </div>
            <div className="card report-card">
              <div className="section1 clearfix">
                <div className="col2 last">
                  <div className="grid clearfix">
                    <div className="col3 first">
                      <h1>
                        <img
                          className="party-logo"
                          src={PdpLogo}
                          alt="APC logo"
                        />
                      </h1>
                      <span>Following</span>
                    </div>
                    <div className="col3">
                      <h1>452</h1>
                      <span>Members</span>
                    </div>
                  </div>
                </div>
              </div>
              <strong>Party Name: </strong>
              <div className="report-status resolved">
                People Democratic Party (PDP)
              </div>
              <div className="participant">
                <strong>Aspiring Candidate: </strong>
              </div>
              <p className="report-user">
                <strong>Peter Obi</strong>
              </p>
              <strong>Political office: </strong>
              <p className="report-time">Senator</p>
              <strong>Party Headquarter:</strong>
              <p className="report-location">4b Wuse Abuja, Nigeria</p>
              <strong>Registered:</strong>
              <p className="report-title">21st Nov. 2018 10:00 am</p>
              <p className="report-media">Political Interest</p>
            </div>
            <div className="card report-card">
              <div className="section1 clearfix">
                <div className="col2 last">
                  <div className="grid clearfix">
                    <div className="col3 first">
                      <h1>
                        <img
                          className="party-logo"
                          src={PartyLogo}
                          alt="APC logo"
                        />
                      </h1>
                      <span>Following</span>
                    </div>
                    <div className="col3">
                      <h1>452</h1>
                      <span>Members</span>
                    </div>
                  </div>
                </div>
              </div>
              <strong>Party Name: </strong>
              <div className="report-status resolved">
                All Progressives Congress (APC)
              </div>
              <div className="participant">
                <strong>Aspiring Candidate: </strong>
              </div>
              <p className="report-user">
                <strong>David Mark</strong>
              </p>
              <strong>Political office: </strong>
              <p className="report-time">House Of Rep</p>
              <strong>Party Headquarter:</strong>
              <p className="report-location">4b Wuse Abuja, Nigeria</p>
              <strong>Registered:</strong>
              <p className="report-title">21st Nov. 2018 10:00 am</p>
              <p className="report-media">Political Interest</p>
            </div>
          </div>
        </section>
      </main>
    </div>
  );
};
export default CandidateList;
