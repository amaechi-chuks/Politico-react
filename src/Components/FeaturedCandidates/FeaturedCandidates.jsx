import React from 'react';
import { Link } from 'react-router-dom';
import '../../assets/style/howItWorks.css';
import PartyLogo from '../../assets/img/PartyLogo.png';
import PdpLogo from '../../assets/img/PdpLogo.png';

const FeaturedCandidates = () => {
  return (
    <div>
      <section className="featured">
        <h2 className="section-title">Featured Candidates</h2>
        <div className="column featured-cards">
          <div className="card report-card">
            <div className="section1 clearfix">
              <div className="col2 last">
                <div className="grid clearfix">
                  <div className="col3 first">
                    <h1>
                      <img
                        className="party-logo"
                        src={PartyLogo}
                        alt="APC logo"
                      />
                    </h1>
                    <span>Following</span>
                  </div>
                  <div className="col3">
                    <h1>452</h1>
                    <span>Members</span>
                  </div>
                </div>
              </div>
            </div>
            <strong>Party Name: </strong>
            <div className="report-status resolved">
              All Progressives Congress (APC)
            </div>
            <div className="participant">
              <strong>Aspiring Candidate: </strong>
            </div>
            <p className="report-user">
              <strong>Muhammadu Buhari</strong>
            </p>
            <strong>Political office: </strong>
            <p className="report-time">President</p>
            <strong>Party Headquarter:</strong>
            <p className="report-location">4b Wuse Abuja, Nigeria</p>
            <strong>Registered:</strong>
            <p className="report-title">21st Nov. 2018 10:00 am</p>
            <i className="report-media" />
            Political Interest
          </div>
          <div className="card report-card">
            <div className="section1 clearfix">
              <div className="col2 last">
                <div className="grid clearfix">
                  <div className="col3 first">
                    <h1>
                      <img
                        className="party-logo"
                        src={PartyLogo}
                        alt="APC logo"
                      />
                    </h1>
                    <span>Following</span>
                  </div>
                  <div className="col3">
                    <h1>452</h1>
                    <span>Members</span>
                  </div>
                </div>
              </div>
            </div>
            <strong>Party Name: </strong>
            <div className="report-status resolved">
              All Progressives Congress (APC)
            </div>
            <div className="participant">
              <strong>Aspiring Candidate: </strong>
            </div>
            <p className="report-user">
              <strong>Prof. Yemi Osibanjo</strong>
            </p>
            <strong>Political office: </strong>
            <p className="report-time">Vice President</p>
            <strong>Party Headquarter:</strong>
            <p className="report-location">Plot 5 Wuse Abuja, Nigeria</p>
            <strong>Registered:</strong>
            <p className="report-title">21st Jan. 2016 10:00 am</p>
            <i className="report-media" />
            Political Interest
          </div>
          <div className="card report-card">
            <div className="section1 clearfix">
              <div className="col2 last">
                <div className="grid clearfix">
                  <div className="col3 first">
                    <h1>
                      <img
                        className="party-logo"
                        src={PdpLogo}
                        alt="PDP logo"
                      />
                    </h1>
                    <span>Following</span>
                  </div>
                  <div className="col3">
                    <h1>692</h1>
                    <span>Members</span>
                  </div>
                </div>
              </div>
            </div>
            <strong>Party Name: </strong>
            <div className="report-status resolved">
              People Democratic Party (PDP)
            </div>
            <div className="participant">
              <strong>Aspiring Candidate: </strong>
            </div>
            <p className="report-user">
              <strong>Atiku Abubakar</strong>
            </p>
            <strong>Political office: </strong>
            <p className="report-time">President</p>
            <strong>Party Headquarter:</strong>
            <p className="report-location">Plot 5 Wuse Abuja, Nigeria</p>
            <strong>Registered:</strong>
            <p className="report-title">21st Oct. 2017 10:00 am</p>
            <i className="report-media" />
            Political Interest
          </div>
          <div className="card report-card">
            <div className="section1 clearfix">
              <div className="col2 last">
                <div className="grid clearfix">
                  <div className="col3 first">
                    <h1>
                      <img
                        className="party-logo"
                        src={PdpLogo}
                        alt="APC logo"
                      />
                    </h1>
                    <span>Following</span>
                  </div>
                  <div className="col3">
                    <h1>692</h1>
                    <span>Members</span>
                  </div>
                </div>
              </div>
            </div>
            <strong>Party Name: </strong>
            <div className="report-status resolved">
              People Democratic Party (PDP)
            </div>
            <div className="participant">
              <strong>Aspiring Candidate: </strong>
            </div>
            <p className="report-user">
              <strong>Atiku Abubakar</strong>
            </p>
            <strong>Political office: </strong>
            <p className="report-time">President</p>
            <strong>Party Headquarter:</strong>
            <p className="report-location">Plot 5 Wuse Abuja, Nigeria</p>
            <strong>Registered:</strong>
            <p className="report-title">21st Oct. 2017 10:00 am</p>
            <i className="report-media" />
            Political Interest
          </div>
          <Link to="view-more-candidate" className="btn btn-primary2">
            View More Candidatess
          </Link>
        </div>
      </section>
    </div>
  );
};
export default FeaturedCandidates;
