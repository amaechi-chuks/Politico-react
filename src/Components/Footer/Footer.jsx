import React from 'react';

const Footer = () => {
  return (
    <div>
      <footer>
        <p className="credits">
          Politico | Copyright &copy; Amaechi-Chuks&nbsp;
          {new Date().getFullYear()}
        </p>
      </footer>
    </div>
  );
};

export default Footer;
