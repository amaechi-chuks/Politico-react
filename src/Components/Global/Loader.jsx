import React from 'react';
import '../../assets/style/global/Loader.css';

const Loader = () => {
  return (
    <div className="loader-container">
      <p className="loader" />
    </div>
  );
};

export default Loader;
