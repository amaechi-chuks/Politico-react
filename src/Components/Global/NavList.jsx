import React from 'react';
import PropTypes from 'prop-types';

const NavList = props => {
  return (
    <React.Fragment>
      <li className="navbar-link" key={props.name}>
        {props.children}
      </li>
    </React.Fragment>
  );
};
NavList.propTypes = {
  children: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};
export default NavList;
