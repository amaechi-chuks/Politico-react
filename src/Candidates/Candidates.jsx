import React from 'react';
import CandidateList from './CandidateList';
import Header from '../Components/Header/Header';

const Candidates = () => {
  return (
    <div>
      <Header />
      <CandidateList />
    </div>
  );
};
export default Candidates;
