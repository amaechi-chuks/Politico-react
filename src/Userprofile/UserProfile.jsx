/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import Notifications, { notify } from 'react-notify-toast';
import UserHeader from '../Components/Global/User';
import avatar from '../assets/img/avatar.png';
import upload from '../services/upload';
import Loader from '../Components/Global/Loader';
import ProfileTable from './ProfileTable';
import Button from '../Components/Global/Buttons';

class UserProfile extends Component {
  constructor(props) {
    super(props);
    this.state = {
      imageUrl: null,
      loading: false,
    };
  }

  handleChange = async e => {
    this.setState({ loading: true });
    const file = e.target.files[0];
    const reader = new FileReader();
    reader.readAsDataURL(file);
    reader.onload = () => {
      this.setState({ imageUrl: reader.result });
    };
  };

  uploadImage = async () => {
    const { imageUrl } = this.state;
    const formData = new FormData();
    formData.append('passporturl', imageUrl);
    await upload.uploadPic(formData);
  };

  render() {
    const user = JSON.parse(localStorage.getItem('user'));
    const { loading } = this.state;
    return (
      <div>
        {loading && <Loader />}
        <Notifications />
        <UserHeader />
        <section className="container">
          <h2 className="section-title">User Profile</h2>
          <div className="profile">
            <div className="card profile-card">
              <img
                className="profile-image"
                src={user.passporturl === null ? avatar : user.passporturl}
                alt="avatar"
              />
              <div className="card-container">
                <h4>
                  <b>
                    {user.firstname}
                    &nbsp;
                    {user.lastname}
                  </b>
                </h4>
                <p>{user.phonenumber}</p>
                <p>{user.email}</p>
                <form encType="multipart/form-data">
                  <p>Change Profile Picture:</p>
                  <label htmlFor="file-upload" className="custom-file-upload">
                    <i className="fa fa-upload" />
                    Upload
                    <input
                      type="file"
                      id="file-upload"
                      onChange={this.handleChange.bind(this)}
                    />
                    <Button
                      onClick={this.uploadImage}
                      value="Upload!"
                      className="btn btn-primary user-form"
                    />
                  </label>
                </form>
              </div>
            </div>
            <ProfileTable />
          </div>
        </section>
      </div>
    );
  }
}

UserProfile.propTypes = {};

export default UserProfile;
