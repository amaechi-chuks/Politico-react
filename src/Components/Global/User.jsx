/* eslint-disable jsx-a11y/anchor-is-valid */
/* eslint-disable jsx-a11y/anchor-has-content */
/* eslint-disable react/no-this-in-sfc */
import React from 'react';
import { Link } from 'react-router-dom';
import NavList from './NavList';
import '../../assets/style/howItWorks.css';
import AppLogo from '../../assets/img/AppLogo.png';

const UserHeader = () => {
  return (
    <div>
      <div className="modal-toggle" />
      <header className="sticky-top">
        <nav className="navbar">
          <Link to="/" className="navbar-logo">
            <img src={AppLogo} alt="App logo" />
          </Link>
          <button type="submit" className="hamburger-menu">
            <a className="hamburger" />
          </button>
          <ul className="navbar-menu">
            <li className="navbar-link">
              <Link to="/">Home</Link>
            </li>
            <li className="navbar-link">
              <Link to="/apply">Apply to contest</Link>
            </li>
            <li className="navbar-link">
              <Link to="./vote.html">Vote</Link>
            </li>
            <li className="navbar-link active">
              <Link to="/profile">Profile</Link>
            </li>
            <li className="navbar-link">
              <Link to="/view-more-candidate">View Candidates</Link>
            </li>
            <li className="navbar-link">
              <Link to="./index.html">Log Out</Link>
            </li>
          </ul>
        </nav>
      </header>
    </div>
  );
};
export default UserHeader;
