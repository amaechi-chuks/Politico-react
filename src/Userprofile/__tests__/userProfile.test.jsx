import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import PartyList from '../../Party/PartyList/PartyList';
import OfficeList from '../../Office/OfficeList/OfficeList';
import UserProfile from '../UserProfile';
import ReviewerRequestCard from '../../Party/PartyCards/PartyCards';
import CandidateList from '../../Candidates/CandidateList';
import Candidates from '../../Candidates/Candidates';
import FeaturedCandidates from '../../Components/FeaturedCandidates/FeaturedCandidates';
import Footer from '../../Components/Footer/Footer';
import Button from '../../Components/Global/Buttons';
import Input from '../../Components/Global/Inputs';
import Loader from '../../Components/Global/Loader';
import NavList from '../../Components/Global/NavList';
import User from '../../Components/Global/User';
import Header from '../../Components/Header/Header';
import HowItWorks from '../../Components/HowItWorks/HowItWorks';
import LandingPage from '../../Components/LandingPage/LandingPage';

const office = {
  name: 'federal',
  type: 'senate',
};
const party = {
  name: 'chuks',
  logoUrl: 'hjkwejheknwjkc',
  hqAddress: '3 bode thomas',
};
const props = {
  fetchAllParty: jest.fn(),
  fetchAllOffice: jest.fn(),
};

const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const store = mockStore({});

const ProfilePageCont = (
  <Provider store={store}>
    <UserProfile {...props} />
  </Provider>
);

describe('ProifilePage component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(ProfilePageCont);
    expect(wrapper).toMatchSnapshot();
  });

  // it('should render userPrfile page without crashing', () => {
  //   localStorage.setItem('user', '{ "id": 2, "passporturl": "url" }');
  //   const wrapper = shallow(<UserProfile {...props} />);
  //   expect(wrapper.find('div'));
  // });

  it('should render party without crashing', () => {
    const wrapper = shallow(<PartyList {...party} />);
    expect(wrapper.find('div'));
  });

  it('should render ReviewedArticles without crashing', () => {
    const wrapper = shallow(<OfficeList {...office} />);
    expect(wrapper.find('div'));
  });
});
describe('<PROFILE CARD />', () => {
  it('should render profile card', () => {
    const wrapper = shallow(
      <ReviewerRequestCard initials="" logoUrl="" hqAddress="" name="" />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div'));
  });
});
describe('<CANDIDATES LIST CARD />', () => {
  it('should render candidates list card', () => {
    const wrapper = shallow(<CandidateList />);
    expect(wrapper.find('div'));
  });
});
describe('<CANDIDATES CARD />', () => {
  it('should render candidates card', () => {
    const wrapper = shallow(<Candidates />);
    expect(wrapper.find('div'));
  });
});
describe('<REQUEST CARD />', () => {
  it('should render request card', () => {
    const wrapper = shallow(<FeaturedCandidates />);
    expect(wrapper.find('div'));
  });
});
describe('<FOOTER />', () => {
  it('should render footer', () => {
    const wrapper = shallow(<Footer />);
    expect(wrapper.find('div'));
  });
});
describe('<BUTTON />', () => {
  it('should render button', () => {
    const wrapper = shallow(<Button />);
    expect(wrapper.find('div'));
  });
});
describe('<FORM INPUT />', () => {
  it('should render form input', () => {
    const wrapper = shallow(<Input />);
    expect(wrapper.find('div'));
  });
});
describe('<LOADER />', () => {
  it('should render loader', () => {
    const wrapper = shallow(<Loader />);
    expect(wrapper.find('div'));
  });
});
describe('<NAVLIST/>', () => {
  it('should render nvalist section', () => {
    const wrapper = shallow(<NavList />);
    expect(wrapper.find('div'));
  });
});
describe('<USER SECTION />', () => {
  it('should render user section', () => {
    const wrapper = shallow(<User />);
    expect(wrapper.find('div'));
  });
});
describe('<HEADER SECTION  />', () => {
  it('should render header section', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find('div'));
  });
});
describe('<HOWITWORKS SECTION />', () => {
  it('should render howitworks section', () => {
    const wrapper = shallow(<HowItWorks />);
    expect(wrapper.find('div'));
  });
});
describe('<LANDINGPAGE SECTION />', () => {
  it('should render landing page section', () => {
    const wrapper = shallow(<LandingPage />);
    expect(wrapper.find('div'));
  });
});
