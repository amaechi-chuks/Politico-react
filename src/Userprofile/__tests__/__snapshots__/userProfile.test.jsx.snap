// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<REQUEST CARD /> should render request card 1`] = `ShallowWrapper {}`;

exports[`AdminPage component should match snapshot 1`] = `ShallowWrapper {}`;
